### PR TITLE
Upsample support NHWC

### PR DESCRIPTION
**Pull request by yihonglyu**
Implement bilinear interpolation for Upsample (Resize) 4-D input with the
outermost and innermost scale (usually channel of NHWC) as 1.

Besides, I revert the HandleResize back to the original implementation for
TransposeOptimizerTests.TestResize* tests.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/tensor/upsample.cc [hunk 1]
```diff
@@ -1065,22 +1066,65 @@ Status Upsample<T>::BaseCompute(OpKernelContext* context,
...
+          input_height = dims[0];
+          input_width = dims[1];
+
+          output_height = output_dims[0];
+          output_width = output_dims[1];
+
+          height_scale = scales[0];
+          width_scale = scales[1];
+        } else {
+          if (scales[1] == 1.0f) {
```
Comment (askhade) [inline onnxruntime/core/providers/cpu/tensor/upsample.cc:1097]: Is this a good enough indicator to determine the layout of the tensor?

**Comment by askhade** [comment]
What is the motivation for this change?
Layout transformer considers this op as layout sensitive because cpu and nnapi treat this as a layout senstive op. Changes are required in layout transformer to handle the case where resize may be in nhwc format. see https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc#L852

File: onnxruntime/core/providers/cpu/tensor/upsample.cc (hunk 1 above)
Comment (yihonglyu) [reply onnxruntime/core/providers/cpu/tensor/upsample.cc:1097]: Before entering this method, ScalesValidation is invoked.
```
      ORT_ENFORCE(scales.size() == 2 ||
                      (scales.size() == 4 && scales[0] == 1 && scales[1] == 1) ||
                      (scales.size() == 4 && scales[0] == 1 && scales[3] == 1) ||
                      scales.size() == 3 ||
                      (scales.size() == 5 && scales[0] == 1 && scales[1] == 1),
                  "'Linear' mode only support:\n"
                  "  * 2-D inputs or\n"
                  "  * 3-D inputs ('Bilinear', 'Trilinear') or\n"
                  "  * 4-D inputs with the corresponding outermost 2 scale values being 1"
                  " or the corresponding outermost and innermost scale values being 1 or\n"
                  "  * 5-D inputs with the corresponding outermost 2 scale values being 1"
                  "in the ",
                  is_resize_ ? "Resize operator" : "Upsample operator");
```
So this is enough to determine the layout of the tensor.

**Comment by yihonglyu** [comment]
> What is the motivation for this change? Layout transformer considers this op as layout sensitive because cpu and nnapi treat this as a layout senstive op. Changes are required in layout transformer to handle the case where resize may be in nhwc format. see https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/optimizer/transpose_optimizer/optimizer_api_impl.cc#L852

This patch is to handle https://github.com/microsoft/onnxruntime/issues/9857. To work around the issue, the commit
```
commit 16bfd3c771c417fa047aa0e551dc493818c2bb4c
Author: RandySheriffH <48490400+RandySheriffH@users.noreply.github.com>
Date:   Mon Nov 29 22:30:16 2021 -0800

    Cancel transpose optimizer for resize (#9870)
```
comment out the transpose logic temporally. With your change 
```
commit f436d3437e85fa0c28a6264dd361958ae4d63e08
Author: Ashwini Khade <askhade@microsoft.com>
Date:   Tue Feb 15 20:25:29 2022 -0800

    Add layout transformer for NNAPI (#10371)
```
All TransposeOptimizerTests.Test* test:
```
1: [ FAILED ] TransposeOptimizerTests.TestResize
1: [ FAILED ] TransposeOptimizerTests.TestResizeOpset11
1: [ FAILED ] TransposeOptimizerTests.TestResizeOpset15
1: [ FAILED ] TransposeOptimizerTests.TestResizeSizeRoi
1: [ FAILED ] TransposeOptimizerTests.TestResizeRoiScalesZeroRank0
1: [ FAILED ] TransposeOptimizerTests.TestResizeNonconst
1: [ FAILED ] TransposeOptimizerTests.TestResizeNonconstOpset13
```
fails even enabling transpose optimizer for resize. Could you give me some insight about it?

**Comment by yihonglyu** [comment]
After revert the HandleResize back to original implementation, all tests passes.

File: onnxruntime/core/providers/cpu/tensor/upsample.cc (hunk 1 above)
Comment (yufenglee) [reply onnxruntime/core/providers/cpu/tensor/upsample.cc:1097]: could you please add comment here to avoid this kind of confusion?

**Pull request merged**